File: InstantSearchApp/japp/src/main/java/com/woowrale/instantsearch/japp/ui/search/remote/RemoteSearchView.java
package com.woowrale.instantsearch.japp.ui.search.remote;

import com.woowrale.instantsearch.japp.data.model.Contact;
import com.woowrale.instantsearch.japp.ui.adapters.ContactsAdapter;
import com.woowrale.instantsearch.japp.ui.base.BasePresenter;

import java.util.List;

import io.reactivex.subjects.PublishSubject;

public interface RemoteSearchView extends BasePresenter.View {

    ContactsAdapter getmAdapter();

    List<Contact> getContactsList();

    PublishSubject<String> getPublishSubject();

}

File: AndroidPatternsApp/kmvp-app/src/main/java/com/woowrale/kmvp/ui/search/remote/RemoteSearchView.kt
package com.woowrale.kmvp.ui.search.remote

import com.woowrale.kmvp.data.model.Contact
import com.woowrale.kmvp.ui.adapters.ContactsAdapter
import com.woowrale.kmvp.ui.base.BasePresenter
import io.reactivex.subjects.PublishSubject

interface RemoteSearchView : BasePresenter.View {

    fun getmAdapter(): ContactsAdapter

    fun getContactsList(): List<Contact>

    fun setContactsList(contacts: List<Contact>)

    fun getPublishSubject(): PublishSubject<String>
}